### PR TITLE
Migrate FerretDB to Kong

### LINKED ISSUE #1156: Migrate FerretDB and envtool to Kong
Just like #1103, but for other binaries.

Migrating FerretDB will allow us to provide environment variables for configuration in addition to flags.

- [x] #1190
- [ ] #1184

### PR DESCRIPTION
**Pull request by noisersup**
# Description

This PR closes #1156.

## Readiness checklist

* [x] I added tests for new functionality or bugfixes.
* [x] I ran `task all`, and it passed.
* [x] I added/updated comments for both exported and unexported top-level declarations (functions, types, etc).
* [ ] I checked comments rendering with `task godocs`.
* [x] I ensured that the title is good enough for the changelog.
* [x] (for maintainers only) I set Reviewers ([`@FerretDB/core`](https://github.com/orgs/FerretDB/teams/core)), Assignee, Labels, Project and project's Sprint fields.
* [x] I marked all done items in this checklist.

### CODE REVIEW

**Comment by noisersup** [comment]
```
> go run ./cmd/ferretdb --help
Usage: ferretdb

Flags:
  -h, --help                                                            Show context-sensitive help.
      --version                                                         Print version to stdout (full version, commit, branch, dirty flag) and exit.
      --listen-addr="127.0.0.1:27017"                                   Listen address.
      --proxy-addr="127.0.0.1:37017"                                    Proxy address.
      --debug-addr="127.0.0.1:8088"                                     Debug address.
      --mode="normal"                                                   Operation mode: [normal proxy diff-normal diff-proxy].
      --test-record=""                                                  Directory of record files with binary data coming from connected clients.
      --handler="pg"                                                    Backend handler: dummy, pg.
      --postgresql-url="postgres://postgres@127.0.0.1:5432/ferretdb"    PostgreSQL URL.
      --log-level="debug"                                               Log level: debug, info, warn, error.
      --test-conn-timeout=0                                             Test: set connection timeout.
```
```
> go run -tags=ferretdb_tigris ./cmd/ferretdb --help
Usage: ferretdb

Flags:
  -h, --help                                                            Show context-sensitive help.
      --version                                                         Print version to stdout (full version, commit, branch, dirty flag) and exit.
      --listen-addr="127.0.0.1:27017"                                   Listen address.
      --proxy-addr="127.0.0.1:37017"                                    Proxy address.
      --debug-addr="127.0.0.1:8088"                                     Debug address.
      --mode="normal"                                                   Operation mode: [normal proxy diff-normal diff-proxy].
      --test-record=""                                                  Directory of record files with binary data coming from connected clients.
      --handler="pg"                                                    Backend handler: dummy, pg, tigris.
      --postgresql-url="postgres://postgres@127.0.0.1:5432/ferretdb"    PostgreSQL URL.
      --log-level="debug"                                               Log level: debug, info, warn, error.
      --test-conn-timeout=0                                             Test: set connection timeout.
      --tigris-client-id=""                                             Tigris Client ID.
      --tigris-client-secret=""                                         Tigris Client secret.
      --tigris-token=""                                                 Tigris token.
      --tigris-url="http://127.0.0.1:8081/"                             Tigris URL.
```
```
> go run ./cmd/envtool --help
Usage: envtool

Flags:
  -h, --help     Show context-sensitive help.
      --debug    Enable debug mode.
```

**Comment by AlekSi** [comment]
@noisersup Please merge with `main`

**Review by AlekSi — COMMENTED** [review]
Let's try something like that:

```go
var cli struct {
  // ...
}

func main() {
  initCLI()
  run()
}

func initCLI() {
  kong.Parse(&cli, ...)
  // ...
}

func run() {
  // main logic that uses populated `cli`
}

func TestMain(m *testing.M) {
 // magic
}
```

I'm not sure what the magic is yet, but not calling `kong.Parse` or any other kong code inside `run()` should be the first step

File: cmd/ferretdb/main_test.go [hunk 1]
```diff
@@ -17,17 +17,41 @@
...
 // TestMain ensures that command-line flags are initialized correctly when FerretDB is run with coverage enabled.
 func TestMain(m *testing.M) {
-	//initCLI()
+	// Split flags for kong and for `go test` by "--". For example:
+	// bin/ferretdb-testcover -test.coverprofile=cover.txt -- --test-record=records --mode=diff-normal --listen-addr=:27017
+	// forKong: --test-record=records --mode=diff-normal --listen-addr=:27017
+	// forTest: -test.coverprofile=cover.txt
+	forKong := os.Args[1:]
+	forTest := []string{}
+	i := slices.Index(os.Args, "--")
```
Comment (noisersup) [inline cmd/ferretdb/main_test.go:41]: Complex problems require simple solutions :) I

**Review by w84thesun — DISMISSED** [review]
LGTM

**Review by AlekSi — CHANGES_REQUESTED** [review]
> I added/updated comments for both exported and unexported top-level declarations (functions, types, etc).

That is not done

File: cmd/ferretdb/main.go [hunk 2]
```diff
@@ -117,13 +124,13 @@ func main() {
 
 	var found bool
 	for _, m := range clientconn.AllModes {
-		if *modeF == string(m) {
+		if cli.Mode == string(m) {
```
Comment (AlekSi) [inline cmd/ferretdb/main.go:127]: I think we could refactor it by using `slices.Contains`

**Comment by noisersup** [comment]
> That is not done

Should we comment `run()`?

**Comment by AlekSi** [comment]
> Should we comment run()?

> "unexported top-level declarations (functions, [...])"

Yes, we should :)

**Review by AlekSi — APPROVED** [review]

**Review by w84thesun — APPROVED** [review]

**Pull request merged**